assets/src/web-stories-block/block/components/storiesInspectorControls.js

```diff
@@ -0,0 +1,256 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+  TextControl,
+  PanelBody,
+  RangeControl,
+  SelectControl,
+  ToggleControl,
+  Notice,
+} from '@wordpress/components';
+import { InspectorControls } from '@wordpress/block-editor';
+import { createInterpolateElement, useEffect } from '@wordpress/element';
+import { select } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig } from '../../../dashboard/app/config';
+import {
+  CAROUSEL_VIEW_TYPE,
+  CIRCLES_VIEW_TYPE,
+  GRID_VIEW_TYPE,
+  ORDER_BY_OPTIONS,
+} from '../constants';
+import { isShowing } from '../util';
+import AuthorSelection from './authorSelection';
+
+/**
+ * StoriesInspectorControls props.
+ *
+ * @typedef StoriesInspectorControlsProps
+ *
+ * @property {string} viewType String indicator of active view type.
+ * @property {number} numOfStories Number indicator of maximum number of stories to show.
+ * @property {number} numOfColumns Number indicator of number of columns in grid view type.
+ * @property {string} orderByValue String indicator of stories sorting.
+ * @property {boolean} isShowingTitle Whether or not to display story's title.
+ * @property {boolean} isShowingDate Whether or not to display story's date.
+ * @property {boolean} isShowingAuthor Whether or not to display story's author.
+ * @property {boolean} isShowingViewAll Whether or not to display stories archive link.
+ * @property {string} viewAllLinkLabel Archive link's label.
+ * @property {boolean} imageOnRight Whether or not to display images on right side in list view type.
+ * @property {Array} authors An array of authors objects which are currently selected.
+ * @property {Function} setAttributes Callable function for saving attribute values.
+ */
+
+/**
+ * LatestStoriesBlockControls component. Used for rendering block controls of the block.
+ *
+ * @param {StoriesInspectorControlsProps} props Component props.
+ *
+ * @return {*} JSX markup.
+ */
+const StoriesInspectorControls = (props) => {
+  const {
+    attributes: {
+      viewType,
+      numOfStories,
+      numOfColumns,
+      orderByValue,
+      viewAllLinkLabel,
+      authors,
+      sizeOfCircles,
+      fieldState,
+    },
+    setAttributes,
+    showFilters = true,
+  } = props;
+
+  const { fieldStates } = useConfig();
+
+  useEffect(() => {
+    // Set default field state on load.
+    if (!Object.entries(fieldState).length) {
+      setAttributes({
+        fieldState: fieldStates,
+      });
+    }
+  }, [fieldState, fieldStates, setAttributes]);
+
+  // Set up sort options.
+  const orderByOptions = Object.entries(ORDER_BY_OPTIONS).map(
```

Comment:
Again this could use a useMemo.

Reply:
I believe these are not computationally that heavy methods that we can see direct impact. Though I'll go through the block code to see if we have any major computations that could use `useMemo`